src/main/java/de/hpi/bpt/argos/eventProcessing/mapping/EventEntityMapperImpl.java

```diff
@@ -119,4 +81,46 @@ private String getAttributeValue(long typeAttributeId, List<Attribute> attribute
 
 		throw new EventEntityMappingException("attribute not found");
 	}
+
+	/**
+	 * This method builds a mapping sql query.
+	 * @param mappingConditions - the list of all mapping conditions to consider in the query
+	 * @param eventAttributes - the list of all attribute of the received event
+	 * @return - the mapping sql query, or an empty string if something went wrong
+	 */
+	private String buildMappingQuery(List<MappingCondition> mappingConditions, List<Attribute> eventAttributes) {
+		if (mappingConditions.isEmpty()) {
+			return "";
+		}
+
+		try {
+			StringBuilder sqlWhere = new StringBuilder();
+			sqlWhere.append("WHERE");
```

Comment:
In my opinion it would be better to have the WHERE keyword in the actual query and just build the rest of the clause here.